extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerMemoryParameters.java

```diff
@@ -439,11 +494,19 @@ private static long memoryNeededForInputChannels(final int numInputWorkers)
   }
 
   /**
-   * Amount of heap memory available for our usage.
+   * Amount of heap memory available for our usage. Any computation changes done to this method should also be done in its corresponding method {@link WorkerMemoryParameters#estimateTotalMemoryInJvmFromUsableMemory}
+   */
+  private static long computeUsableMemoryInJvm(final long maxMemory, final long totalLookupFootprint)
+  {
+    return (long) ((maxMemory - totalLookupFootprint) * USABLE_MEMORY_FRACTION);
```

Comment:
What does `USABLE_MEMORY_FRACTION` represent? Should it apply to lookups? That is, if `USABLE_MEMORY_FRACTION` represents memory given over to Java, process overhead, etc., then we should apply that to total memory, and then subtract lookups, since lookups are subject to the same overhead as worker memory. Else, maybe explain the reasoning.

Reply:
SInce lookups can be really big and will only have a hashmap/hashmaps the overhead would be trivial hence IMO the current logic is fine. Added this comment there.